### PR TITLE
Update docs and rules for Choclatey moderation

### PR DESCRIPTION
**Pull request by BiggerNoise**
Fixes most of #274 . Everything but the docs\LICENSE and docs\NOTICE needing to be removed.

On that one I have tried to create a build.custom/copy_notices_to_doc_folder.step, but it doesn't appear that my custom step got picked up. @ferventcoder  - I'm going to kick that last bit over to you.

While you're there, please edit/approve my VERIFICATION.txt verbiage.

### CODE REVIEW

**Comment by erikbra** [comment]
(If you merge #285, you can do this easily directly in the .nuspec file).

File: nuget/roundhouse/tools/VERIFICATION.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+RoundhousE is from the Chuck Norris organization, and we maintain the package.
```
Comment (ferventcoder) [inline nuget/roundhouse/tools/VERIFICATION.txt]: 👍

File: build.custom/nugetPrepare.post.step [hunk 2]
```diff
@@ -69,6 +69,7 @@
         <include name="buildCustomization/**" />
         <include name="logo/**" />
         <include name="*.dot" />
+        <include name="*.docx" />
```
Comment (ferventcoder) [inline build.custom/nugetPrepare.post.step]: 👍

**Comment by erikbra** [comment]
@BiggerNoise How should we treat this and #274 ? Are we happy? The builds off AppVeyor should pass Chocolatey moderation now, but maybe we should wait for the Cake scripts of @ferventcoder  before we submit anything? Just wanted to update the "Getting started" here: https://github.com/chucknorris/roundhouse (had a fellow developer download RH, and I had to search for where to D/L it - better to update the docs, but it would be nice to just say 

> cinst roundhouse

to new devs.

**Comment by BiggerNoise** [comment]
You have mentioned that if we did a git-flow type branching, that GitVersion would have an easier time properly labeling our versions. Did I understand that correctly?

Wonder if it might make sense to adopt that so we can produce properly versioned artifacts with a bit less manual effort.

As far as resolving this issue, I would need to push a pre-release to choclatey. I don't think that's a big deal, but I will probably not be able to do it this week (we're re-arranging the house, painting a couple of rooms, it's not an ideal time for concentration).

Bigger issue is if we cut a release 0.8.9 or 0.9.0 with the current state of the software, we're going to mess with the Oracle people (until we can get the managed library in place). Given my distractions, this probably means that we would be breaking Oracle in a couple of weeks.

Given that the issue with Oracle is well known and that you have a P/R that covers it, I can live with telling any broken Oracle user to use 0.8.8 or to help us get it fixed. I'm not willing to indefinitely delay the next release until somebody from that community steps up to help us finish the work.

Thought? @ferventcoder - I'd appreciate your input as well.

**Comment by ferventcoder** [comment]
I wouldn't say I was an expert in all of the database types. I did make an effort to learn and test all of them though to be sure they were working. I bet AV (appveyor) can give us that coverage for testing.

If there are known breakages, I will be a stickler that they should be clearly communicated and documented with mitigation strategies in the release notes. That is the minimum effort. I would def like to see those things get fixed as soon as possible for oracle folks as I'm not a fan of regressions. It does take a bit to stand up an environment to test such a thing though.

**Comment by erikbra** [comment]
I don't think we have a problem with Oracle in master as of now. We just get warnings that `System.Data.OracleClient`, that we use, is obsolete. So I tried to move it to the supported, managed Oracle driver, which is the PR. But master is working, I think.

The thing with gitflow and git version, is that is nicely fixes the version numbers. It appends `-unstable`, e.g. if we were in the develop branch now, the version would be `0.8.9-unstable` (i think this is changed to `-alpha` in Gitversion4) to all versions if you are in the develop branch. When you create a release branch, it appends `-beta.1`. When you tag with a release, say `0.8.9`, then the version number is a pure `0.8.9`. It does work better with using `,0` in the end for "major releases", and .1, .2, .3 for "minor" ones.

We could then publish directly to e.g. MyGet unstable builds, and beta and stable ones to nuget. Maybe this would help us get some traction on testing e.g. Oracle? I don't know how many use it with Oracle. Maybe we should have a poll, or add some telemetry?

If you're not familiar with GitFlow, i think  Atlassians explanation here is brilliant (nice pictures): https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow. We are using it at work, and to me, at least, it works. But we don't use PRs so much there. It kinda looks like Forking workflow should not use branches in the main repo (at least per Atlassian, https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow), but I see advantages with having a `develop` and a `master` branch. PRs are made agains develop, and testing and stuff is done there. When we think we're happy, we create a release branch (or not - maybe no point), do final testing an verification, tag with the release number, and release.  Then we merge back to master (if we did a release branch), and develop.

**Comment by BiggerNoise** [comment]
Supposedly, the  issue that we will have with Oracle with what is in master is that we have switched to using AnyCPU which apparently has some issues with the Oracle native binary. Several issues and P/Rs have mentioned this, I have no direct experience with the issue myself. By way of remediation, the Oracle folks can continue to use 0.8.8 which is an x86 binary.

Thanks for the links on Gitflow. That's what we use at work as well. I have noticed that processes tend to mutate a bit to match their environments, so it's good to see that what they describe there matches what we do. 

Do you mind setting that up to work with AppVeyor? I can certainly set up the dev branch, but I'll be a little lost on the AppVeyor side of things. I don't think my 30 second creation of a develop branch buys us that much.

**Comment by erikbra** [comment]
Yes, I can look at the AppVeyor stuff, no problem. Although, I don't think
it's that much to do. If you just name the branch `develop`, I think a lot
of the version numbers are fixed by themselves. Should we try that? That we
create a develop now, and try to make PRs agains that one, and merge into
Master (or a release branch) when we're ready for release? Do any of you
have any experience with MyGet? I could of course create an account, I see
people tend to use that for non-official Nuggets (I know they're called
Nugets, I just think nuggets sound better ;)).

2017-11-09 14:48 GMT+01:00 Andy Davis <notifications@github.com>:

**Comment by ferventcoder** [comment]
Branching strategies - this is likely a much bigger discussion. The tl;dr is that you release from tags only, there be pain and confusion in any other method.

**Comment by ferventcoder** [comment]
For GitVersion and MyGet, you can get a lot by just mirroring exactly what @gep13 has set up for Chocolatey GUI - https://github.com/chocolatey/chocolateygui

**Comment by erikbra** [comment]
Thanks for the tip, @ferventcoder. But, as you see, ChocolateyGui also has a develop branch, and that is the one they're building in CI. But, it looks like GitVersion is very configurable, it's just that it works with a GitFlow flow without any configuration. I'll look into how we can get it to work with only a master branch, http://gitversion.readthedocs.io/en/latest/configuration/

**Comment by erikbra** [comment]
@ferventcoder and @BiggerNoise, You do agree that this GitHubFlow (first time I've heard of it, but I'm not a very seasoned GitHub-user) describes how we develop RoundhousE, right?

http://gitversion.readthedocs.io/en/latest/git-branching-strategies/githubflow/

**Comment by ferventcoder** [comment]
Apologies, I wasn't referring to the branch names, but the concept of not deploying when you merge to master. It is a confusing concept for folks and rife with error. A tag is much more representative of a release.

I honestly don't mind if we want a branch named "develop" and one named "master",  the names don't matter but the concept of a branch where current work is occurring and something stable for tagging releases is a great idea. Just don't deploy/release based on merging to that stable branch, only do so based on a tag creation. 

In chocolatey/choco we use master/stable as long-lived branches, but that could be develop/master (stable). The idea is one for new dev and verification, and one that is representative of current deployment plus fixes.

**Comment by ferventcoder** [comment]
ChocolateyGUI is a great example of what you may be trying to achieve ad something to emulate, and I was just pointing out it could be a good one to look at versus attempting to reinvent the wheel.

**Comment by erikbra** [comment]
I created a PR, #300, which versions nicely (at least IMHO) when using only master branch and PRs. We might of course switch to using to branches, one unstable, and one stable (whatever we choose to call them), but it is not necessary, as I thought previously.

If we wish (I think it would be a good idea) we could now get the nupkgs off AppVeyor to MyGet. Who of you has the Myget/F/roundhouse account?

**Comment by gep13** [comment]
@BiggerNoise @erichmx let me know if you have any questions about GitVersion, MyGet, etc.  Happy to talk at length about each of these 😄

**Comment by erikbra** [comment]
Sorry, guys. This thread has moved in all directions :) (I am very guilty of this). If we merge #300 we can tag master with 0.8.9, and create a release. I think that one's good enough for Chocolatey moderation. 

@BiggerNoise do you feel the x86/AnyCpu issue for Oracle peeps should be solved before a release? Or should Oracle users stick with 0.8.8? What's the biggest drawback? Do we know anyone running on anything other than x86? We can't run on ARM, can we? Maybe we should just change it back to x86?

**Comment by BiggerNoise** [comment]
We had several P/Rs and requests to support AnyCPU. I am inclined to grant that higher consideration than Oracle support. 

I'd certainly like to support them going forward, but someone from that community is going to have to step up and help us get the managed library working.

**Comment by BiggerNoise** [comment]
I merged #300, but just don't have the bandwidth to do a release this week. (Still home doing room painting/relocation.)

If @ferventcoder wants to plug you in to the NuGet & Choclatey accounts, I would completely support making a new release. Since we're going to be (possibly) breaking Oracle, let's do two things:
1. Mark it up in the release notes.
2. Call the release 0.9.0

I should be past this renovation stuff in a week or so and hopefully able to contribute a bit more.

**Comment by BiggerNoise** [comment]
[We're in](https://chocolatey.org/packages/roundhouse/0.9.0)

**Pull request closed without merging**